vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -92,69 +107,48 @@ public ImageData getImageDimensions(IVSImage image, int width, int height) {
-        pageNo = validatePageNumber(pageNo);
-        if(sortedBy!=null && SortByField.getAllValues().contains(sortedBy)) {
-            sortingParameters = Sort.by(sortedBy);
-        }
-        if(order!=null && order.equalsIgnoreCase(Sort.Direction.ASC.toString())) {
-            sortingParameters = sortingParameters.ascending();
-        }
-        else {
-            sortingParameters = sortingParameters.descending();
-        }
+    public List<IVSImage> getImages(int pageNo, String category, String sortedBy, String order) {
+        Sort sortingParameters = getSortingParameters(sortedBy, order);
+        pageNo = validatePageNumber(pageNo, category);
         Pageable sortByRequestedField = PageRequest.of(pageNo - 1, pageSize, sortingParameters);
-        Page<VSImage> images = imageRepo.findAll(sortByRequestedField);
+        Page<VSImage> images;
+        if(category.equals(Constants.ALL)) {
+            images = imageRepo.findAll(sortByRequestedField);
+        }else {
+            images = imageRepo.findByCategories(sortByRequestedField, ImageCategory.valueOf(category));
+        }
         List<IVSImage> results = new ArrayList<>();
         if(images != null) {
             images.getContent().forEach(i -> results.add(i));
         }
         return results;
     }
 
-    /**
-     * Method to return the total pages sufficient to display all images
-     * 
-     * @return totalPages required to display all images in DB
-     */
     @Override
-    public long getTotalPages() {
-        return (imageRepo.count() % pageSize==0) ? imageRepo.count() / pageSize:(imageRepo.count() / pageSize) + 1;
+    public long getTotalImageCount(String category) {
+        if(category.equals(Constants.ALL)) {
+            return imageRepo.count();
+        }
+        return imageRepo.countByCategories(ImageCategory.valueOf(category));
     }
 
-    /**
-     * Method to return the total image count
-     * 
-     * @return total count of images in DB
-     */
     @Override
-    public long getTotalImageCount() {
-        return imageRepo.count();
+    public long getTotalPages(String category) {
+        if(category.equals(Constants.ALL)) {
+            return (imageRepo.count() % pageSize==0) ? imageRepo.count() / pageSize:(imageRepo.count() / pageSize) + 1;
+        }
+        long count = imageRepo.countByCategories(ImageCategory.valueOf(category));
+        return (count%pageSize==0) ? count/pageSize : (count/pageSize)+1;
     }
 
-    /**
-     * Method to return page number after validation
-     * 
-     * @param pageNo page provided by calling method
-     * @return 1 if pageNo less than 1 and lastPage if pageNo greater than
-     *         totalPages.
-     */
     @Override
-    public int validatePageNumber(int pageNo) {
-        long totalPages = getTotalPages();
+    public int validatePageNumber(int pageNo, String category) {
+        long totalPages = getTotalPages(category);
```

Comment:
Is there a reason you deleted all javadoc?  it definitely needs some.

Reply:
I know this method was already here, but does it need to be public? Is it used anywhere outside of this class?

Reply:
?

Reply:
This an open question.

Reply:
The method is also used in Controller and hence cannot be made private.